src/OrchardCore.Modules/OrchardCore.ContentFields/Views/DateField-Localized.Edit.cshtml

```diff
@@ -2,36 +2,25 @@
 
 @using OrchardCore.ContentFields.Settings;
 @using OrchardCore.ContentManagement.Metadata.Models
-@using OrchardCore.ResourceManagement
 
 @inject IResourceManager ResourceManager
 
 @{
     var settings = Model.PartFieldDefinition.Settings.ToObject<DateFieldSettings>();
-    AppendDatePickerScript();
-}
-
-@functions{
-    private void AppendDatePickerScript()
-    {
-        const string cultureSeparator = "-";
-        var culture = Orchard.CultureName();
-        var language = culture.Contains(cultureSeparator)
-            ? culture.Substring(0, culture.IndexOf(cultureSeparator))
-            : culture;
-        var datePickerScript = new TagBuilder("script");
-        datePickerScript.InnerHtml.AppendHtml($@"$(function () {{
-            $('.datepicker').datepicker($.datepicker.regional['{language}']);
-        }});");
-
-        ResourceManager.RegisterFootScript(datePickerScript);
-    }
-}
+    var language = Orchard.CultureName().Split('-')[0];
+    var datepickerId = Html.IdFor(m => m.Value);
+}       
 
 <fieldset class="form-group">
     <div class="row col-md-6 col-lg-4">
         <label asp-for="Value">@Model.PartFieldDefinition.DisplayName()</label>
         <input asp-for="Value" type="text" class="form-control content-preview-select datepicker" required="@settings.Required" />
         <span class="hint">@settings.Hint</span>
     </div>
-</fieldset>
\ No newline at end of file
+</fieldset>
+
+<script at="Foot">
+$(function () {
+    $('#@datepickerId').datepicker($.datepicker.regional['@language']);
```

Comment:
Just one comment, why we didn't use `datepicker` class instead?

Reply:
It would call it multiple times if you had multiple fields. The ideal solution would be to have an external file using the class, then registering the file would happen only once.

Reply:
> It would call it multiple times if you had multiple fields. 

I expect this will register once per page even if there 're multiple fields

Reply:
Nope, I can take a look at how the RM should be changed to handle this case. Maybe just by giving a name to a <script> tag, and maybe it's already how it works ;)